infra/src/main/kotlin/fi/fta/geoviite/infra/authorization/AuthorizationController.kt

```diff
@@ -17,16 +15,13 @@ import org.springframework.security.access.prepost.PreAuthorize
 import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.web.bind.annotation.*
 
-@RestController
+@GeoviiteController
 @RequestMapping("/authorization")
```

Comment:
Muistaakseni olin itsekin juuri tämän mallin, nimeämisten jne. puolella kun näistä juteltiin viime viikolla, mutta nyt kun näitä lukee käytännössä niin alankin kääntää kelkkaani... En olekaan enää ihan varma onko tämä annotaatioiden yhdisteleminen lopulta hyvä juttu, sillä se kadottaa eksplisiittisen tiedon siitä, että pohjimmiltaan kyseessä on kumminkin Springin `@RestController`-/`@Service`-annotaatiot - asia, joka saattaa olla yllättävän tärkeä silloin kun koodipesä, Spring tai tällainen annotaatiomagia on vielä ei-niin-tuttuja. Lisäksi kun nuo meidän omat annotaatiot on nimetty noin niinkuin ovat, niin niistä ei ole ollenkaan selvää mitä ne oikeasti tekevät (`Geoviite` on aika geneerinen etuliite Geoviite-nimisessä softassa.) Hiukan pyöriteltyäni tätä mielessäni näkisin tässä kolmisen eri reittiä parempaan maailmaan:
1. Jätetään erilliset annotaatiot controllereille, serviceille jne., otetaan niistä annotaatioiden yhdistely pois ja nimetään ne siten että niistä ilmenee että niiden elämäntehtävä on lokittaa (esim. `@ControllerLogging`/@ServiceLogging`/`@DaoLogging` tms.) Tämän myötä esim. Controllerien annotaatiot tulisivat käytännössä menemään `@RestController @ControllerLogging @RequestMapping("urli")`, servicejen `@Service @ServiceLogging` jne.
2. Jätetään yhdistely elämään, mutta uudelleennimetään nuo uudet annotaatiot kuvaamaan lokitustehtävää (`@LoggingController`/`@LoggingService` jne.) Mietitään nimeämisiä uudestaan sitten jos niille tulee joskus muitakin vastuualueita.
3. Mennään yhdistelyssä päätyyn asti ja yhdistetään Controllereissa tuo `@RequestMapping`-annotaatio mukaan `@GeoviiteController`:iin, jolloin sillä on yli yksi vastuualue ja siten joku muu kuin `Logging`:iin viittaava nimi on oikeutettu.

Oman ääneni saa siis tuo 1.-vaihtoehto, sillä se on lukijalle selkeästi kaikkein eksplisiittisin. Mutta ehkä tästä voisi jutella vielä dailyssä tms. että saadaan muidenkin mielipiteitä asiaan.

Reply:
Jeps eli tämä muutettu tavan 3. mukaisesti kuten aamulla jutskattiin. GeoviiteController sisältää nyt siis myös tuon pohjaurlin määrityksen ja käyttöpaikat on muovattu sen mukaisesti. Osa testicontrollereista (+testeistä) edellytti hieman muutoksia, sillä ne eivät asettaneet pohjaurlia samalla tavalla kuin varsinaisen controllerit.